### DISCUSSION TITLE
Running Gotenberg Locally

### DISCUSSION THREAD

**Discussion by Lincoln-hub** [gotenberg/gotenberg#440] [Q&A]
I am running Gotenbergr on my local computer, I installed all the required engines but I'm running into this error: [FATAL] starting uno: start long-running LibreOffice listener: execute LibreOffice listener: context done: context deadline exceeded exit status 1.  Does anyone know how to fix this?

**Comment by gulien** [comment]
Did you try the latest version, i.e., `7.5.1`?

**Reply by Lincoln-hub** [reply]
Thank you for your quick response and help, I tried the latest version, and everything is working except libreoffice/convert. It's giving this error:
ERROR   api     convert to PDF: lock long-running LibreOffice listener: start long-running LibreOffice listener: execute LibreOffice listener: context done: context deadline exceeded.

**Reply by gulien** [reply]
Could you add the debug logs? 🤔

**Reply by Lincoln-hub** [reply]
Again, thanks for the quick response.

2022/03/23 16:37:21.279 ERROR   api     convert to PDF: lock long-running LibreOffice listener: start long-running LibreOffice listener: execute LibreOffice listener: context done: context deadline exceeded        {"trace": "4ac3cb61-c6bf-4338-832a-c47f8f801d79", "remote_ip": "::1", "host": "localhost:3000", "uri": "/forms/libreoffice/convert", "method": "POST", "path": "/forms/libreoffice/convert", "referer": "", "user_agent": "PostmanRuntime/7.28.4", "status": 503, "latency": 10004247290, "latency_human": "10.00424729s", "bytes_in": 9026, "bytes_out": 19}

Are these the debug logs?

**Reply by gulien** [reply]
Actually, you have to launch the app with `--log-level=debug` 😄

**Reply by Lincoln-hub** [reply]
Thank you for the tip I learned something new today.
2022/03/24 16:40:54.287 DEBUG   api.formslibreofficeconvert     form data values: map[] {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:40:54.287 DEBUG   api.formslibreofficeconvert     form data files: map[Data_for_scaling.xlsx:/tmp/49881073-1a0c-4f4b-bed5-55dbc3981e14/Data_for_scaling.xlsx]   {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:40:54.287 DEBUG   api.formslibreofficeconvert     LibreOffice listener lock acquired      {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:40:54.287 DEBUG   api.formslibreofficeconvert     starting LibreOffice listener...        {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:40:54.287 DEBUG   api.formslibreofficeconvert.usrbinlibreoffice   start unix process: /usr/bin/libreoffice --headless --invisible --nocrashreport --nodefault --nologo --nofirststartwizard --norestore -env:UserInstallation=file:///tmp/e511dfc3-2ab2-4f05-9eed-8ea738382c5b --accept=socket,host=127.0.0.1,port=40047,tcpNoDelay=1;urp;StarOffice.ComponentContext {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:41:04.289 DEBUG   api.formslibreofficeconvert.usrbinlibreoffice   unix process killed     {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:41:04.289 DEBUG   api.formslibreofficeconvert     '/tmp/49881073-1a0c-4f4b-bed5-55dbc3981e14' removed     {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec"}
2022/03/24 16:41:04.289 ERROR   api     convert to PDF: lock long-running LibreOffice listener: start long-running LibreOffice listener: execute LibreOffice listener: context done: context deadline exceeded        {"trace": "00ab4337-0858-4418-940c-09fbbffd9dec", "remote_ip": "::1", "host": "localhost:3000", "uri": "/forms/libreoffice/convert", "method": "POST", "path": "/forms/libreoffice/convert", "referer": "", "user_agent": "PostmanRuntime/7.28.4", "status": 503, "latency": 10002593154, "latency_human": "10.002593154s", "bytes_in": 9026, "bytes_out": 19}

**Reply by Lincoln-hub** [reply]
And also I apologize I am  new to Go so the structure of the source code is a bit complicated for me. I want to use this a service on my computer where I download a file from a s3 bucket convert the file to pdf  and upload it. Is there a way to directly call the functions that do the conversions without having to go through the API.  For example instead of going through the http://localhost:3000/forms/libreoffice/convert, I can just call the function that do the conversions directly from the another module. Thank you again for your help.

**Reply by gulien** [reply]
@Lincoln-hub you may first increase the start timeout with `--uno-listener-start-timeout=30s` 😄 hopefully, that would do the trick

**Reply by Lincoln-hub** [reply]
@gulien Thank you for your help. I'm still having the same error even after increasing the start timeout. It seems the libreoffice process is being killed right after it starts.

**Reply by gulien** [reply]
I have no idea why it does not work.. 🤔

**Reply by gulien** [reply]
I’ve created issue #442 to follow your issue. May also be related to #420.
